content/ideas/web-scraping-monad.md

```diff
@@ -0,0 +1,16 @@
+---
+title: Make Web scraping monad that is easier than state-of-art solution
+---
+
+Web scraping is of large practical importance for machine learning, search engines, and many other big data applications.
+
+We already have Haxl monad for optimizing asynchronous data gathering from known sources, but our scraping capabilities in Haskell are much poorer than in other languages like Python ScraPy.
+
+It would be great to implement special web-scraping monad, and equip it with many commonly used facilities, like authentication simulation, web front-end testing, and download acceleration.
+
+(There are many more features, that I can enumerate to interested student.)
+
+**Mentor**: [Michal Gajda](mailto:migamake&#64;migamake.com)
+
+**Difficulty**: Advanced
+
```

Comment:
Something like this already exists: [Scalpel](https://hackage.haskell.org/package/scalpel).

We always prefer for GSoC projects to contribute to existing projects rather than adding new ones, since it increases the chance of the code finding a happy home.  If the Scalpel project is not in a sufficient state right now, then this proposal should focus on enumerating the missing bits and having a student contribute those to that project.